File: manual/code/generate_strategies/IR(pseudocode)/cpp/cpp/function_sortedcontainers_SortedDict_popitem.cpp
#include <string>
#include <sstream>
#include <boost/container/flat_map.hpp>

class SortedDictHandler {
public:
    std::string modify_sorted_dict(int index) {
      
        boost::container::flat_map<int, std::string> sorted_dict;
        sorted_dict.emplace(3, "three");
        sorted_dict.emplace(1, "one");
        sorted_dict.emplace(5, "five");

        try {
          
            if (index < 0) {
               
                index = sorted_dict.size() + index;
            }
            
            if (index < 0 || index >= static_cast<int>(sorted_dict.size())) {
                return "error: Invalid index";
            }

       
            auto it = sorted_dict.begin();
            std::advance(it, index);

      
            auto key = it->first;
            auto value = it->second;
            std::pair<int, std::string> removed_item = std::make_pair(key, value);

        
            sorted_dict.erase(it);

      
            std::ostringstream result;
            result << "{";
            bool first = true;
            for (const auto& pair : sorted_dict) {
                if (!first) {
                    result << ", ";
                }
                result << pair.first << ": '" << pair.second << "'";
                first = false;
            }
            result << "} (" << removed_item.first << ", '" << removed_item.second << "')";

            return result.str();
        }
        catch (...) {
            return "error: Invalid index";
        }
    }
};

File: code/generate_strategies/IR(CoT)/java/java/function_sortedcontainers_SortedDict_popitem.java
package com.example;

import com.google.common.collect.Maps; 
import java.util.Map;
import java.util.SortedMap;
import org.apache.commons.lang3.tuple.ImmutablePair; 
import org.apache.commons.lang3.tuple.Pair;

public class SortedDictHandler {
    public String modify_sorted_dict(int index) {
        
        SortedMap<Integer, String> sortedDict = Maps.newTreeMap();
        sortedDict.put(3, "three");  
        sortedDict.put(1, "one");    
        sortedDict.put(5, "five");   

        
        if (index < 0) {
            index = sortedDict.size() + index;
        }
        
        if (index < 0 || index >= sortedDict.size()) {
            return "error: Invalid index";
        }

        
        int cur = 0;
        Integer keyToRemove = null;
        String valueToRemove = null;
        for (Map.Entry<Integer, String> entry : sortedDict.entrySet()) {
            if (cur == index) {
                keyToRemove = entry.getKey();
                valueToRemove = entry.getValue();
                break;
            }
            cur++;
        }

        
        sortedDict.remove(keyToRemove);
        
        Pair<Integer, String> removedItem = new ImmutablePair<>(keyToRemove, valueToRemove);

        
        StringBuilder dictString = new StringBuilder();
        dictString.append("{");
        int count = 0;
        for (Map.Entry<Integer, String> entry : sortedDict.entrySet()) {
            if (count > 0) {
                dictString.append(", ");
            }
            dictString.append(entry.getKey()).append(": '").append(entry.getValue()).append("'");
            count++;
        }
        dictString.append("}");

        
        String removedItemString = "(" + removedItem.getLeft() + ", '" + removedItem.getRight() + "')";

        return dictString.toString() + " " + removedItemString;
    }
}